test/extensions/quic_listeners/quiche/platform/epoll_address_test_utils_impl.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <sys/socket.h>
+
+#include <algorithm>
+
+#include "envoy/network/address.h"
+
+#include "test/test_common/environment.h"
+
+namespace epoll_server {
+
+namespace {
+
+int addressFamilyUnderTestHelper() {
+  std::vector<Envoy::Network::Address::IpVersion> versions =
+      Envoy::TestEnvironment::getIpVersionsForTest();
+  if (std::find(versions.begin(), versions.end(), Envoy::Network::Address::IpVersion::v4) !=
+      versions.end()) {
+    return AF_INET;
+  }
+  if (std::find(versions.begin(), versions.end(), Envoy::Network::Address::IpVersion::v6) !=
+      versions.end()) {
+    return AF_INET6;
+  }
+  return -1;
+}
+
+} // namespace
+
+int AddressFamilyUnderTestImpl() {
```

Comment:
Looks like this is confusingly named, but doing the right thing.
Possibly worth a rename or comments that this returns the "highest" supported address version of v4/v6.  Given it's test code and I suspect will be unused I'm fine landing as-is.

Reply:
Renaming requires upstream change. I added a comment here for the return value.